README.md

```diff
@@ -22,9 +26,10 @@ The connector is great for getting started with Conduit but also for certain typ
 
 ### Configuration
 
-| name          | description                                                                                | required | Default |
-|---------------|--------------------------------------------------------------------------------------------|----------|---------|
-| recordCount   | Number of records to be generated. -1 for no limit.                                        | false    | "-1"    |
-| readTime      | The time it takes to 'read' a record.                                                      | false    | "0s"    |
-| fields        | A comma-separated list of name:type tokens, where type can be: int, string, time, bool.    | true     | ""      |
-| format        | Format of the generated payload data: raw, structured.                                     | false    | "raw"   |
+| name        | description                                                                                                                                        | required | Default | Example                                                                                                 |
+|-------------|----------------------------------------------------------------------------------------------------------------------------------------------------|----------|---------|---------------------------------------------------------------------------------------------------------|
+| recordCount | Number of records to be generated. -1 for no limit.                                                                                                | false    | "-1"    | 15 (generates 15 records)                                                                               |
+| readTime    | The time it takes to 'read' a record.                                                                                                              | false    | "0s"    | 100ms (generates records every 100ms)                                                                   |
+| fields      | A comma-separated list of name:type tokens, <br/>where type can be: int, string, time, bool.                                                       | true     | ""      | id:int,name:string <br/> (generates a struct with an ID field, type int, and a name field, type string) |
+| format      | Format of the generated payload data: raw, structured.<br/> If fields are specified, then the payload will be a JSON string's byte representation. | false    | "raw"   |                                                                                                         |
```

Comment:
Something crossed my mind just now: since we're talking about three mutually exclusive options, maybe we should simply have 3 mutually exclusive config. params, e.g.:
1. `raw.file`
2. `raw.fields`
3. `structured.fields`?

Reply:
That would mean we need to add another config option every time we add another format 😕 I'd rather have 2 fields that are static and we add new accepted values for those fields.

> What do you suggest be in `format.options`, simply a string, which the connector would deserialize?

The value has to be a string (it's a `map[string]string` after all). The value would be deserialized based on the `format.type` - if it's `raw` or `structured` it should be the comma-separated tokens (as before), if it's `file` it should be the path to a file. That's how I see it.

Reply:
> That would mean we need to add another config option every time we add another format confused I'd rather have 2 fields that are static and we add new accepted values for those fields.

Yes, you're right there! And from the aspect of the config size, format.type + format.options is definitely better. I'm also thinking: 

If we're going to add many more config. options, then we will push that all complexity into the `format.options` string. It makes it slightly more complex for us to parse (versus the other solution). It does make it slightly more complex for the user too (because they need to carefully change the string).

On the other hand, if we're doing to add only a couple of more options (which IMO is the more likely case), I think that's still manageable. Also, at least for now, we're not combining them, so for the time being, a user would be able to use just one (e.g. `raw.file: /folder/file.txt` vs `format.type: file` and `format.options: /folder/file.txt`).

That said, IMHO, the "decomposed" config. options appear to be doing well for both (us, when parsing, users, when configuring), so I still prefer it. I don't see any option radically different or better than the other one, so I'm fine implementing either in the interest of time.: )

Reply:
Honestly, I much prefer the approach with 2 options, where what you choose for the first option dictates what you should put into the second one. I feel it's more intuitive and even a more common pattern for the user.

I don't think it will be more complex to parse - on the contrary, I think it would be easier. We would check the first option (`format.type`) and see if it contains a valid value, based on that we choose the strategy to parse the second option (`format.options`).

I also think it is surprising to see multiple options that are incompatible to use at the same time - when I see a design like this I always think to myself "why didn't they structure it in a way where I couldn't even provide both options at the same time?". That's what the 2 field option accomplishes. Combined with the fact that this approach gives us the ability to easily add more options to generate different payloads in the future without adding new config fields, I feel quite strongly this is the way to go.

Reply:
I'll go with the 2 options option then.: ) Also, for the sake of a shared understanding and idea exchange:

> where what you choose for the first option dictates what you should put into the second one. I feel it's more intuitive and even a more common pattern for the user.

That's a fair point. Specifying one of the types is a good thing for sure. We have actually two things: choosing an option (explicitly vs implicitly) and configuring that option.

What I'm having most problems with is options string, which is the thing I believe is more complex to parse. It necessitates a format (likely our own), users have to be more careful about it etc. As options are added, it becomes bigger and more complex (e.g. we may want  to add a key or metadata to the generated payload, because it triggers certain actions, with OpenCDC we may want to specify the before and after values etc.). A single string can also be ambiguous, and you can for example have format=raw, options=id:int,name:string. You can change the format to file, it will continue working, but it will be writing to an unexpected file.

It generally becomes difficult working with configs which are simply string to string maps, which is why I like structured configs more. And if not possible, I prefer to see some structure in the keys more then dealing with parsing the values.

Reply:
I'm also a fan of having structured configs. The problem here is that the structure is unknown in advance, each connector prescribes its own config structure, it's just simpler for Conduit to operate with a map and let the connector simulate the structure with `.` in config fields. (Sidenote: this gave me an idea - in pipeline config files we could allow the user to actually specify the options in a structured way and we would flatten the structure by joining nested keys with a `.`)

I understand your concern about the option value getting complex, but it's entirely in our hands to prevent this from happening. If in the future we would want to let the user choose a format for the key or metadata I would argue for introducing a separate config field rather than making the options more complex (e.g. `payload.format.type/options`, `key.format.type/options` and `metadata.format.type/options`). As for the argument about changing the `type` and leaving the `options` as they are - well as long as the options make sense for the other type that's fine. Though in your example the validation would actually fail because `name:type` tokens are not a valid file path. But if they were, the file wouldn't exist. But if it does, then all is well 😄

The main quarrel I have is actually with mutually exclusive config fields, to me that is simply unexpected behavior. I am used to dependent options (e.g. `tls.enabled` needs to be true for `tls.certificate` to be considered). I am not used to multiple fields that require me to populate only/exactly one of them, otherwise making the config invalid.

That said, you nicely pointed out that choosing an option explicitly is one side of the coin, the other is configuring that option. In our case, I would still argue that having 1 field for options is simpler and good enough. I would resort to the approach with separate options in case the option string actually gets too complex, but I definitely wouldn't start with it, since, you know, YAGNI 🙂